### PR TITLE
Correct probability_distribution of class Source

### PR DESCRIPTION
**Pull request by MathiasPnt**
probability_distribution of class Source for v0.7.3. The class Source now includes:
- occupation factor
- multiphoton component
- overall_transmisison = losses * brightness

### CODE REVIEW

File: perceval/components/source.py [hunk 1]
```diff
@@ -21,66 +21,123 @@
...
-            svd[StateVector([1], {0: ["_:%s" % random_feat]})] = distinguishability*self.brightness*self.purity
-            svd[StateVector([1], {0: ["_:0"]})] = (1-distinguishability)*self.brightness*self.purity
+                if self._multiphoton_model == "distinguishable":
+                    svd[StateVector([2], {1: {"_": 0}, 2: {"_": noise_photon}})] = eta**2*p2
+                else:
+                    svd[StateVector([2])] = eta**2*p2
+
+        if distinguishability != 0:
+            if self._multiphoton_model == "distinguishable":
+                svd[StateVector([1], {1: {"_": distinguishable_photon}})] = eta*distinguishability*p1+eta*(1-eta)*distinguishability*p2
```
Comment (Alex-PLACET) [inline perceval/components/source.py:126]: l.126 and l.130 are the same. You can put it outside the condition scope

File: perceval/components/source.py [hunk 2]
```diff
@@ -21,66 +21,123 @@
...
+                multiphoton_component = 0
+            else:
+                multiphoton_component = 1 - purity
+        else:
+            assert purity is None, "cannot set both purity and multiphoton_component"
+            
+        if occupation_factor is None:
+            if brightness is None:
+                ocupation_factor = 1
+        else:
```
Comment (Alex-PLACET) [inline perceval/components/source.py:63]: Warnin: You have an issue in the indentation. Here the else if for l.60

File: perceval/components/source.py [hunk 3]
```diff
@@ -21,66 +21,123 @@
...
-        r""":return: SVDistribution on 1 mode associated to the source
+    def probability_distribution(self):
+        r"""returns SVDistribution on 1 mode associated to the source
         """
+
+        g2 = self.multiphoton_component
+        eta = self.overall_transmission*self.brightness/self.occupation_factor
+        px = self.occupation_factor
+
+        # g2 = 2p2/(p1+2p2)**2
```
Comment (Alex-PLACET) [inline perceval/components/source.py:89]: Do you need to keep these commented lines ?

File: perceval/components/source.py [hunk 4]
```diff
@@ -21,66 +21,123 @@
...
         self.indistinguishability = indistinguishability
         self._indistinguishability_model = indistinguishability_model
         assert self._indistinguishability_model in ["homv", "linear"], "invalid value for indistinguishability_model"
         self._context = context or {}
         if "discernability_tag" not in self._context:
             self._context["discernability_tag"] = 1
 
-    def probability_distribution(self) -> SVDistribution:
-        r""":return: SVDistribution on 1 mode associated to the source
+    def probability_distribution(self):
```
Comment (Alex-PLACET) [inline perceval/components/source.py:81]: This function is a bit big, can you split it in the conditions ?

**Comment by Alex-PLACET** [comment]
Can you add some unit test ?

File: perceval/components/source.py (hunk 2 above)
Comment (Aubaert) [reply perceval/components/source.py:63]: There is also a missing c in occupation factor

Comment (MathiasPnt) [reply perceval/components/source.py:63]: This lines were unnecessary. I found a fix using "eta = self.overall_transmission*self.brightness/self.occupation_factor". I removed them and set occupation_factor=1 and brightness=1 by default.

File: perceval/components/source.py (hunk 3 above)
Comment (MathiasPnt) [reply perceval/components/source.py:89]: This helps understand the physics behind the code. Not necessary but helpful.

File: perceval/components/source.py (hunk 1 above)
Comment (MathiasPnt) [reply perceval/components/source.py:126]: self._multiphoton_model can be either "indistinguishable" or "distinguishable"

File: perceval/components/source.py (hunk 3 above)
Comment (Alex-PLACET) [reply perceval/components/source.py:81]: So please add a comment that explain that is helpul to understand the code.

**Pull request closed without merging**